cmd/osbuild-image-tests/context-managers.go

```diff
@@ -90,7 +91,25 @@ func withBootedQemuImage(image string, ns netNS, f func() error) error {
 			return fmt.Errorf("cannot close temporary cloudinit file: %#v", err)
 		}
 
-		qemuCmd := ns.NamespacedCommand(
+		var qemuCmd *exec.Cmd
+		// This command does not use KVM as I was unable to make it work in Beaker,
+		// once we have machines that can use KVM, enable it to make it faster
```

Comment:
Could we skip the test if we detect that kvm does not work?
I think we should keep the test cases unmodified from the output of the generation scripts, or they will get out-of-sync (and we also want to be able to easily regenerate them without having to edit them), so if we need to skip anything, we should do that here, in the test runner.

Reply:
Another alternative would be to support a commandline option to disable the qemu tests, if auto-detection is not possible.

Reply:
Hmm can you think of a way to detect KVM? I can think of `lsmod | grep kvm`, but I'm not sure how reliable it is.

Reply:
Could we not run `qemu` in a way that forces `kvm` emulation, and if that fails skip the test? Not sure what it allows. @gicmo, @dvdhrm, @larskarlitski: Hilfe!

Reply:
If you use `-M accel=kvm`, it will only use KVM and fail otherwise. If you use `-M accel=kvm:hvf:tcg`, it will try KVM first, then HVF, and lastly TCG. TCG should always work, as it is the built-in ISA emulator IIRC.

I am not sure whether you can really catch the error when KVM is not there. Usually, it is enough to use `test -f /dev/kvm`, though. Might wanna give this a try?

Reply:
`test -f /dev/kvm` sounds ok, but why not just pass `-M accel=kvm:hvf` and skip the test if `qemu` itself fails?

Reply:
How do you know whether qemu failed because of missing KVM or because the test failed? Other than some hacky output-string comparison.

Reply:
I agree with David. If we assume qemu can fail (by only letting it use kvm), let's have a separate check for it by testing for `/dev/kvm`.

`test -f /dev/kvm` will fail. You need `-c` ("FILE exists and is character special") or `-e` ("FILE exists").

Reply:
I thought there would be some way to distinguish between "`qemu` failed during setup" and "invalid image file", but if not, then checking for `/dev/kvm` sounds fine.

Reply:
Is it possible that `/dev/kvm` exists, but `kvm` still does not work? Would `modprobe kvm` be more reliable here?

Reply:
Hmm, I think checking for the device should be sufficient. If that exists but doesn't work, then I think the tests should rightly fail. We may not always be able to modprobe (probably in practice we will, but I don't think we should rely on it).

Reply:
You added the `kvm` detection, so I think the command now should contain `-accel` so that if we do get `kvm` to work it will just work.

Reply:
ok